pkg/services/host.go

```diff
@@ -22,35 +26,73 @@ func (s *HostService) CreateHost(t *domain.Host) (*domain.Host, error) {
 	return s.storage.CreateHost(t)
 }
 
-func (s *HostService) GetHostsByTenantID(tenantID string) ([]*domain.Host, error) {
-
-	// hosts := &[]domain.Host{
-	//
-	// 	{
-	// 		ID:         "1",
-	// 		TenantID:   "fcde6d34-ac73-4f29-8e48-bdc5670e1d69",
-	// 		OperatorID: "74eb9201-2926-40bb-9f8c-a3a52b3b5db7",
-	// 		Value:      "www.facebook.com",
-	// 		Type:       domain.HostType(domain.Domain),
-	// 		CreatedAt:  time.Now().UTC(),
-	// 		UpdatedAt:  time.Now().UTC(),
-	// 	},
-	// 	{
-	// 		ID:         "2",
-	// 		TenantID:   "fcde6d34-ac73-4f29-8e48-bdc5670e1d69",
-	// 		OperatorID: "74eb9201-2926-40bb-9f8c-a3a52b3b5db7",
-	// 		Value:      "www.google.com",
-	// 		Type:       domain.HostType(domain.Domain),
-	// 		CreatedAt:  time.Now().UTC(),
-	// 		UpdatedAt:  time.Now().UTC(),
-	// 	},
-	// }
-
-	hosts, err := s.storage.GetHostsByTenantID(tenantID)
+func (s *HostService) GetHostsByTenantIDAndUserID(tenantID string, userID string) ([]*domain.Host, error) {
+
+	hosts, err := s.storage.GetHostsByTenantIDAndUserID(tenantID, userID)
 
 	if err != nil {
 		return nil, err
 	}
 
 	return hosts, nil
 }
+
+func (s *HostService) GetHostByID(ID string) (*domain.Host, error) {
+	host, err := s.storage.GetHostByID(ID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return host, nil
+}
+
+func (s *HostService) DeleteHostByID(ID string) (bool, error) {
+	isDeleted, err := s.storage.DeleteHostByID(ID)
+
+	if err != nil {
+		return false, err
+	}
+
+	return isDeleted, nil
+}
+
+func (s *HostService) GetHostname(ip_port string) string {
```

Comment:
Again, it seems like you're using this GetHostname method to find out the Domain associated to an IP. On it's own it's useful, but if you're using it to fill out the IP, that's not in scope for our system just yet. I suggest applying security fixes but not getting rid of this function.

Reply:
Remains in code but not enable in api